### PR TITLE
Update Methods.md

### PR DESCRIPTION
**Pull request by matsduf**

### CODE REVIEW

File: docs/specifications/tests/Methods.md [hunk 1]
```diff
@@ -15,46 +15,90 @@ Obtain the parent zone of the given domain (zone).
...
+   Return the name server names.
 
 B. Undelegated test.
 
-2. If the test is an undelegated test, then the name servers names in the input
-   will be used as the equivalent of RDATA of the NS records from the lookup 
-   under A.
+2. Ignore if parent zone or child zone does not exist or cannot be determined.
+
+3. Ignore any name servers provided by delegation from parent zone.
```
Comment (mattias-p) [inline docs/specifications/tests/Methods.md:79]: Steps B.2 and B.3 are noops, right? Have you considered moving them to the description?

File: docs/specifications/tests/Methods.md [hunk 2]
```diff
@@ -15,46 +15,90 @@ Obtain the parent zone of the given domain (zone).
...
 
-2. Send an SOA query for the given domain name (zone) to one of the authoritative
-   name servers for the parent zone.
-3. Collect all A and AAAA records in the additional section and store them 
-   in a cache that method 4 can use.
-4. Collect all NS records in the authority section and return the name server names
-   in the response.
+2. If parent zone could not be determined, method fails.
+
+3. If no child zone exists, method fails.
```
Comment (mattias-p) [inline docs/specifications/tests/Methods.md:70]: Have you considered combining steps A.2 and A.3 into a single condition?

File: docs/specifications/tests/Methods.md [hunk 3]
```diff
@@ -15,46 +15,90 @@ Obtain the parent zone of the given domain (zone).
...
-2. A recursive lookup for the SOA record of the child domain name starting from the
+2. A recursive lookup for the SOA record of the child zone starting from the
    root zone is done, and the steps of the process are recorded.
 3. If the lookup reaches a name server that responds with a redirect (delegation)
-   directly to the requested child zone the method
-   succeeds. The zone in which the delegation was found is
-   considered to be the parent zone.  
+   directly to the requested child zone:
+
+   3.1 The method succeeds.
```
Comment (mattias-p) [inline docs/specifications/tests/Methods.md]: Have you considered using proper lists? That would increase the spacing between the numbering and the content, making it easier on the eye.

File: docs/specifications/tests/Methods.md [hunk 4]
```diff
@@ -15,46 +15,90 @@ Obtain the parent zone of the given domain (zone).
...
+
+3. Ignore any name servers provided by delegation from parent zone.
+
+4. Collect the name server names provided as indata for the test.
+
 3. Collect all IPv4 and IPv6 addresses in the input to the undelegated test and 
    store in a cache where the connection between name server name and IP address
-   is kept. The cache is to be used by method 4.
-
+   is kept. The cache overwrites the cache from method 1 and is to be used by method 4.
```
Comment (mattias-p) [inline docs/specifications/tests/Methods.md:85]: Is the caching stuff related to the previous step?

File: docs/specifications/tests/Methods.md [hunk 5]
```diff
@@ -15,46 +15,90 @@ Obtain the parent zone of the given domain (zone).
...
-2. If the test is an undelegated test, then the name servers names in the input
-   will be used as the equivalent of RDATA of the NS records from the lookup 
-   under A.
+2. Ignore if parent zone or child zone does not exist or cannot be determined.
+
+3. Ignore any name servers provided by delegation from parent zone.
+
+4. Collect the name server names provided as indata for the test.
+
 3. Collect all IPv4 and IPv6 addresses in the input to the undelegated test and
```
Comment (mattias-p) [inline docs/specifications/tests/Methods.md:83]: The numbering is off now.

**Review by mattias-p — CHANGES_REQUESTED** [review]
This is getting detailed. And long.

Have you considered adding a section for inputs? That way inputs could be referenced by name instead of explanation. Second, assumptions about the inputs could be moved from the algorithm section and into the inputs section. Third, I think it makes sense to list the caches that are being used here.

Have you considered also adding a section for state? That way the methods that update caches can declare up front which ones, and the ones that don't can declare that they're stateless.

Have you considered adding a section for return values? That way methods can declare up front whether they can fail or not. Second, descriptions of the meaning of the return value could be moved from the algorithm section and into the return values section. Third, steps that just declare success could be removed as that would be implicit from the fact that the method returns a value.

**Comment by matsduf** [comment]
Not relevant anymore.

**Pull request closed without merging**